akka-stream/src/main/scala/akka/stream/impl/Throttle.scala

```diff
@@ -42,20 +41,21 @@ private[stream] class Throttle[T](cost: Int,
         else completeStage()
 
       override def onPush(): Unit = {
-        val timeElapsed = now() - previousTime
-        val currentTokens = Math.min(timeElapsed * speed + lastTokens, scaledMaximumBurst)
+        val currentTime = now()
+        val currentTokens = Math.min((currentTime - previousTime) * speed + lastTokens, scaledMaximumBurst)
```

Comment:
Ok, finally got it. Adding tokens not on each push but only when it's not enough in bucket.